### PR TITLE
feat(vsc): support for creating a dashboard app in VSC toolkit

### PR DESCRIPTION
**Pull request by huimiu**
- Support for creating a dashboard tab app in VSC. Both `JavaScript` and `TypeScript` languages are supported.

image

- Add the tutorial entry.

image

E2E TEST: N/A

### CODE REVIEW

File: packages/fx-core/src/component/resource/appManifest/utils/ManifestUtils.ts [hunk 1]
```diff
@@ -129,9 +130,15 @@ export class ManifestUtils {
               template.entityId = "index" + staticTabIndex;
               appManifest.staticTabs.push(template);
             } else {
-              const template = cloneDeep(STATIC_TABS_TPL_V3[0]);
-              template.entityId = "index" + staticTabIndex;
-              appManifest.staticTabs.push(template);
+              if (inputs.features && inputs.features === DashboardOptionItem.id) {
```
[the remark this answers is not in the document]
Comment (frankqianms) [reply packages/fx-core/src/component/resource/appManifest/utils/ManifestUtils.ts:133]: elegant

File: templates/package.json [hunk 2]
```diff
@@ -43,7 +43,9 @@
         "tab/js/non-sso",
         "tab/ts/non-sso",
         "tab/js/m365",
-        "tab/ts/m365"
+        "tab/ts/m365",
+        "tab/ts/dashboard",        
+        "tab/js/dashboard"
```
Comment (frankqianms) [inline templates/package.json:48]: nothing serious, but better to swap line 47 & 48, as there is always js first and ts second in this list

File: templates/tab/js/dashboard/src/internal/addNewScopes.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { loginAction } from "./login";
+import FxContextInstance from "./singletonContext";
+import { ErrorWithCode } from "@microsoft/teamsfx";
+
+export async function addNewScope(addscopes) {
```
Comment (frankqianms) [inline templates/tab/js/dashboard/src/internal/addNewScopes.js:5]: please help change the function name to "addNewPermissionScope", along with all the coded refered to this function

File: templates/package.json (hunk 2 above)
Comment (huimiu) [reply templates/package.json:48]: Thanks, I've adjusted their order.

File: packages/fx-core/src/component/resource/appManifest/utils/ManifestUtils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (huimiu) [reply packages/fx-core/src/component/resource/appManifest/utils/ManifestUtils.ts:133]: This is an excellent comment, I've rewritten it.

File: packages/fx-core/src/core/question.ts [hunk 4]
```diff
@@ -358,6 +359,7 @@ export function createCapabilityQuestionPreview(inputs?: Inputs): SingleSelectQu
 
   const staticOptions: StaticOptions = [
     ...newBots,
+    DashboardOptionItem,
     TabNewUIOptionItem,
     TabSPFxNewUIItem,
```
[the remark this answers is not in the document]
Comment (huimiu) [reply packages/fx-core/src/core/question.ts:364]: Yeah, I pasted an extra copy of the picture, I've deleted the extra picture. Thank you for finding this problem.

File: templates/tab/js/dashboard/src/data/ListData.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+[
+    {
+        "id": "id1",
+        "title": "Lorem ipsum",
```
[the remark this answers is not in the document]
Comment (huimiu) [reply templates/tab/js/dashboard/src/data/ListData.json:4]: Both strings have no meaning and are just standard dummy text. For more details on dummy text, please refer to [this link](https://www.lipsum.com/)

File: templates/tab/js/dashboard/src/internal/addNewScopes.js (hunk 3 above)
Comment (huimiu) [reply templates/tab/js/dashboard/src/internal/addNewScopes.js:5]: OK, I've changed the function name and all references.

**Review by eriolchan — APPROVED** [review]

File: templates/tab/js/dashboard/src/internal/addNewScopes.js (hunk 3 above)
Comment (frankqianms) [reply templates/tab/js/dashboard/src/internal/addNewScopes.js]: Maybe the file name should be changed too

**Comment by huimiu** [comment]
As suggested by Zhijie, I will close this PR and split this PR into 3 PRs to make it easier for you to review.

**Pull request closed without merging**